demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+                std::vector<unsigned> results = res.topResults(FLAGS_nt, *completedIRInfo.ir.GetBlob(outputName));
+                std::vector<std::string> predictedLabels = {};
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    bool isPredictionRight = false;
+                    std::string firstPredicted = labels[results[FLAGS_nt*i]];
+                    std::string predictedRight;
+                    for (size_t j = 0; j < FLAGS_nt; j++) {
+                        unsigned predictedClass = results[FLAGS_nt*i+j];
+                        if (predictedClass == rightClasses[i]) {
+                            isPredictionRight = true;
+                            predictedRight = labels[predictedClass];
+                            break;
+                        }
+                    }
+                    if (isPredictionRight) {
+                        predictedLabels.push_back(predictedRight);
+                        rightPredictionsCount++;
+                    }
+                    else {
+                        predictedLabels.push_back(firstPredicted);
+                    }
+                    totalPredictionsCount++;
+
+                    shownImagesInfo.push_back(
+                        std::make_tuple(completedIRInfo.images[i].mat, predictedLabels[i], isPredictionRight));
+                }
+
+                framesNum += FLAGS_b;
+                gridMat.listUpdate(shownImagesInfo);
+
+                avgFPS = 1. / (((cv::getTickCount() - startTickCount) / cv::getTickFrequency()) / framesNum);
+                
+                std::vector<long long> processingEndTimes;
+                cv::Mat showMat = gridMat.getMat(processingEndTimes, hasClasses);
+                if (!FLAGS_no_show && FLAGS_delay != -1) {
+                    cv::imshow("main", showMat);
+                    key = static_cast<char>(cv::waitKey(delay));
+                }
+
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    latencySum += ((processingEndTimes[i] - completedIRInfo.images[i].startTime)
+                                   / cv::getTickFrequency());
+                }
+                avgLatency = latencySum / framesNum;
+                accuracy = static_cast<double>(rightPredictionsCount) / totalPredictionsCount;
+                gridMat.textUpdate(avgFPS, avgLatency, accuracy, isTestMode, hasClasses, presenter);
+            }
+            else if (!emptyInferRequests.empty()) {
+                mutex.unlock();
+                cv::Mat nextImage = inputImages[nextImageIndex];
+                long long IRstartTime = cv::getTickCount();
+                resizeImage(nextImage, modelInputResolution);
+                emptyInferRequests.front().images.push_back(
+                    IRInfo::IRImage(nextImage,
+                                    std::stoul(hasClasses ? classIndexes[nextImageIndex] : "0"),
```

Comment:
It you need to provide a dummy value, you can initially fill `classIndexes` with that values. In that case you won't need to check anything.